docs/softwaredevelopment.md

```diff
@@ -1 +1,124 @@
+
+If your service is an instance of a well-known type of system (like an intranet or help desk), it is appropriate to include the name of the government body it’s associated with so people can understand the context it applies to. Where possible, avoid exposing that name to people (usually citizens or organisations outside government) who shouldn’t have to understand how government is structured to do what they’re trying to do.
+
+Ensure you use the same name consistently whenever you’re referring to the same thing. For example, in most cases the name of an application’s git repository should match its hostname. Similarly, the team responsible for an application should probably be called something related to the application and its git repositories.
+Make sure you use the same name consistently whenever you’re referring to the same product. 
+
+### Other resources
+The [GOV.UK Service Manual](https://www.gov.uk/service-manual) has good guidance for [naming services](https://www.gov.uk/service-manual/design/naming-your-service), a lot of which will also be relevant when naming applications or packages.
+
+## Programming languages
+
+We use a range of programming languages at DDS because we think using the right tool for the job gives us the best chance of building services that best meet users’ needs. This document does not apply to choosing ‘off the shelf’ software (open source or not).
+
+We focus on using a small number of programming languages for core software development tasks.
+
+This should make it easier for developers to:
+
+- move around the organisation
+- develop shared components
+- improve their personal development
+- master how they operate applications
+
+### Frontend development
+
+The GOV.UK Service Manual has information on [using client-side JavaScript](https://www.gov.uk/service-manual/technology/using-progressive-enhancement). 
+
+### Backend development
+
+Our core languages for backend development are:
+
+- Python
+- Go
+
+We’ve chosen these languages because they are successfully used by teams at the moment, and we are confident in how to host and operate applications written in them. You should carry out new development in one of these languages.
+
+#### Python
+You should write new Python projects in Python 3. [Python 2 is end of life](https://www.python.org/doc/sunset-python-2/). Python 3 is now well-supported by application frameworks and libraries, and is commonly used in production.
+
+#### Go
+Go is an appropriate language for systems programming, like proxying, routing, and transforming HTTP requests. However you should only write these sorts of components if there is no alternative maintained open source tool available.
```

Comment:
You already mentioned performance in paragraph below but may here instead.  The performance difference between go and python is vast, however the gap closes with [pypy](https://www.pypy.org/).
I think performance and deployability can take some precedence over the choice that no other open source tool exists (assume this excludes go open source tools and libs?)